doc/mclag/MCLAG_Enhancements_HLD.md

```diff
@@ -0,0 +1,808 @@
+2. MclagSyncd receives the  new MCLAG configuration updates.
+3. MclagSyncd receives new local (both static and dynamic) MAC updates from STATE_DB FDB_TABLE
+4. MclagSyncd sends new MCLAG configuration to ICCPd.
+5. ICCPd sends new remote static MAC updates, ICCP session and MCLAG remote interface state information updates to MclagSyncd.
+6. ICCPd sets new static MAC flag in FDB TLV for local static MAC addresses advertisement. 
+7. MclagSyncd updates ICCP session and MCLAG remote interface state information to STATE_DB FDB_TABLE and MCLAG_REMOTE_INTF_TABLE
+8. MclagSyncd updates MAC addresses learned from peer MCLAG node to new  MCLAG FDB table .
+9. FdbOrch registers for new MCLAG FDB table updates to process MAC updates from peer MCLAG node, ISOGRP Orch process new updates from MclagSyncd.
+10. FdbOrch updates remote MAC addresses with new SAI attribute not to age them out. 
+11. Syncd gets the ICCP learned remote MAC addresses with new SAI attribute SAI_FDB_ENTRY_TYPE_STATIC_MACMOVE set .
+12. Syncd programs the remote MAC to HW
+13. FdbOrch registers with NeighborOrch for all the ICCP learned remote MAC addresses.
+14. NeighborOrch update the MAC address to STATE_DB MCLAG ARP MAC Table.
+15. FdbSycd gets the MAC updates from STATE_DB MCLAG ARP MAC Table.
+16. FdbSyncd programs the MAC updates to kernel.
+
+## 3.2 DB Changes
+
+### 3.2.1 CONFIG DB
+#### 3.2.1.1 MCLAG Domain table
+
+Producer: Configuration
+Consumer: MclagSyncd 
+Description: New table to store MCLAG domain configuration. Applications can use it to get MCLAG configuration
+
+```
+;New MCLAG Domain Table
+key               = MCLAG_DOMAIN|domain_id ; DIGIT 1-4095 
+
+;field             = value
+source_ip          = IP               ; local ip address 
+peer_ip            = IP               ; peer  ip address 
+peer_link          = port_name        ; peer link name
+keepalive_interval = time_in_secs     ; keepalive time in seconds 
+session_timeout    = time_in_secs     ; session timeout in multiple of 
+									  ; keepalive_interval.
+
+;value annotations
+port_name         = 1*64VCHAR        ; name of the port, must be unique
+time_in_secs      = dec-octet        ; time in seconds
+IP                = IPV4address
+IPv4address        = dec-octet "." dec-octet "." dec-octet "." dec-octet "
+
+ dec-octet        = DIGIT            ; 0-9
+                / %x31-39 DIGIT      ; 10-99
+                / "1" 2DIGIT         ; 100-199
+                / "2" %x30-34 DIGIT  ; 200-249
+                / "25" %x30-35       ; 250-255
+```
+#### 3.2.1.2 MCLAG Interface Table
+
+Producer: Configuration
+Consumer: MclagSyncd 
+Description: New table to store MCLAG interfaces for a MCLAG domain. MCLAG applications can query this table to check whether a interface is configured as MCLAG interface
+
+```
+;New MCLAG Interface Table
+key              = MCLAG_INTERFACE|domain_id|ifname ; 
```

Comment:
Could you please add a note here that "ifname" has to be same across the MCLAG Peers. Otherwise, we need to add an intermediate id (ex:mlag-id), to bind the mclag interfaces in all the mclag peers.

Reply:
@prvattem: Thanks for the comment will update accordingly.

Reply:
@prvttem:HLD is updated.